job-service-db/src/main/resources/procedures/taskCollapse.sql

```diff
@@ -0,0 +1,127 @@
+CREATE OR REPLACE FUNCTION task_collapse(tasks VARCHAR[])
+    RETURNS VARCHAR[]
+    LANGUAGE plpgsql
+AS $$
+DECLARE
+    final_array VARCHAR[] :='{}';
+    task VARCHAR;
+    last_nb INTEGER :=0;
+    body VARCHAR :='';
+    counter INTEGER :=2;
+    expected VARCHAR;
+    collapsing BOOLEAN := false;
+    currentT VARCHAR;
+    starting INTEGER:=1;
+    collapsed VARCHAR:='';
+    backup VARCHAR[]:='{}';
+    modified BOOLEAN := 't';
+    maxi INTEGER:=1000;
+    regex_body VARCHAR :='.*(?=\.)\.'; -- takes everything up to the last dot
+    regex_last_nb VARCHAR :='([\d]+)\*?$';-- takes everything after the last dot, excluding the *
+
+BEGIN
+    -- tasks := array_agg(x ORDER BY x DESC) FROM unnest(tasks) x ;
+    SELECT array(
+        SELECT id FROM unnest( tasks ) AS id
+        ORDER BY substring(id,regex_body),
+                 substring(id, regex_last_nb)::INTEGER DESC
+        )
+    INTO tasks;
+
+    -- we loop until the array stop being modified
+    WHILE (modified='t') LOOP
```

Comment:
Can this be simplified?  `== true` is unusual in a while (in general).

Reply:
Done for true